pkg/util/steps/condition.go

```diff
@@ -40,9 +40,11 @@ type conditionStep struct {
 	pollInterval time.Duration
 }
 
-func (c conditionStep) run(ctx context.Context, log *logrus.Entry) error {
+func (c conditionStep) run(ctx context.Context, log *logrus.Entry, timeDilation int) error {
 	var pollInterval time.Duration
-	timeoutCtx, cancel := context.WithTimeout(ctx, c.timeout)
+
+	timeout := time.Duration(c.timeout.Nanoseconds() / int64(timeDilation))
+	timeoutCtx, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
Kind of. It's a "modifier" for how long the timeouts should be, because they're implemented inside the functions. There might be a better option for this.